src/qofonosimwatcher.cpp

```diff
@@ -58,8 +57,12 @@ void QOfonoSimWatcher::Private::onOfonoAvailableChanged()
         allSims.clear();
         if (!presentSims.isEmpty()) {
             presentSims.clear();
-            Q_EMIT watcher->presentSimListChanged();
-            Q_EMIT watcher->presentSimCountChanged();
+            if (watcher) {
+                Q_EMIT watcher->presentSimListChanged();
+            }
+            if (watcher) {
```

Comment:
That is what I initially thought. But it looks like this is handling the case where ~QOfonoSimWatcher is called from the slots connected to the above signals. I would keep the life-time of QOfonoSimWatcher::Private tied to that of QOfonoSimWatcher and require that slots invoked from this object use deleteLater().